### DISCUSSION TITLE
Simulation of animal and human eyes

### DISCUSSION THREAD

**Discussion by davidkh1** [google-deepmind/mujoco#1382] [General]
I am interested in the community's insights regarding MuJoCo's suitability for simulating human and animal eyes in the context of robotic ophthalmic surgery. The aim is to develop models that exhibit physically plausible behaviour, and I am particularly curious whether MuJoCo 3.0 can accurately simulate tissue cutting and fluid injection. 

There are impressive demonstrations of muscle-actuated biomechanics in projects such as OstrichRL and others, but I haven't seen any relevant projects for ophthalmology.

Based on your expertise, do you think it is feasible to develop validated animal and human eye models within the MuJoCo framework? Your insights on this matter would be greatly appreciated.

**Comment by Balint-H** [comment]
MuJoCo can simulate elastic, [jelly like solids](https://github.com/google-deepmind/mujoco/blob/main/model/plugin/elasticity/jelly.xml) but I've never seen it validated against biological soft tissue. Cutting, removing and inserting material is also something that's usually achieved through workarounds (e.g.: https://github.com/google-deepmind/mujoco/issues/197).

MuJoCo would be most suitable to model the robotic part of a process like that. The movement and motion planning of the robotic manipulator can be modelled, and its ability to generate contact forces (e.g. with a needle or blade) of highly specific locations and magnitudes could be evaluated. Basically everything leading up to cutting or injecting tissue.
